test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -83,6 +83,68 @@ public void CreatePackageWithEmptyFoldersForV3Folders()
             }
         }
 
+        [Fact]
+        public void CreatePackageWithFiles()
+        {
+            // Arrange
+            PackageBuilder builder = new PackageBuilder()
+            {
+                Id = "A",
+                Version = NuGetVersion.Parse("1.0"),
+                Description = "Descriptions",
+            };
+
+            builder.Authors.Add("testAuthor");
+
+            var dependencies = new List<PackageDependency>();
+            dependencies.Add(new PackageDependency("packageB", VersionRange.Parse("1.0.0"), null, new[] { "z" }));
+            dependencies.Add(new PackageDependency(
+                "packageC",
+                VersionRange.Parse("1.0.0"),
+                new[] { "a", "b", "c" },
+                new[] { "b", "c" }));
+
+            var set = new PackageDependencyGroup(NuGetFramework.AnyFramework, dependencies);
+            builder.DependencyGroups.Add(set);
+
+            builder.Files.Add(CreatePackageFile(@"build" + Path.DirectorySeparatorChar + "foo.props"));
+            builder.Files.Add(CreatePackageFile(@"content" + Path.DirectorySeparatorChar + "foo.jpg"));
+            builder.Files.Add(CreatePackageFile(@"contentFiles" + Path.DirectorySeparatorChar + "any" + Path.DirectorySeparatorChar + "any" + Path.DirectorySeparatorChar + "foo.png"));
+            builder.Files.Add(CreatePackageFile(@"lib" + Path.DirectorySeparatorChar + "net5.0" + Path.DirectorySeparatorChar + "foo.dll"));
+            builder.Files.Add(CreatePackageFile(@"native" + Path.DirectorySeparatorChar + "net5.0" + Path.DirectorySeparatorChar + "foo"));
+            builder.Files.Add(CreatePackageFile(@"ref" + Path.DirectorySeparatorChar + "net5.0" + Path.DirectorySeparatorChar + "foo.dll"));
+            builder.Files.Add(CreatePackageFile(@"runtimes" + Path.DirectorySeparatorChar + "net5.0" + Path.DirectorySeparatorChar + "foo.dll"));
+            builder.Files.Add(CreatePackageFile(@"tools" + Path.DirectorySeparatorChar + "foo.dll"));
+
+            using (var ms = new MemoryStream())
+            {
+                // Act
+                builder.Save(ms);
+
+                ms.Seek(0, SeekOrigin.Begin);
+
+                using (var archive = new ZipArchive(ms, ZipArchiveMode.Read, leaveOpen: true))
```

Comment:
If you wrote unit tests for `InvalidUndottedFrameworkRule`, you can avoid the CPU intensive work of creating a zip file, just to immediately have it have to re-parse the central directory to find what files are in the zip.  At least the files have zero length content so they don't have to be compressed.

Reply:
Adding a unit test to cover that corner case is a good idea, but I still want the integration test in order to have more general coverage.